src/components/duration/date.rs

```diff
@@ -38,6 +38,292 @@ impl DateDuration {
             days,
         }
     }
+
+    /// 7.5.38 `UnbalanceDateDurationRelative ( years, months, weeks, days, largestUnit, plainRelativeTo )`
+    pub(crate) fn unbalance_relative<C: CalendarProtocol>(
+        &self,
+        largest_unit: TemporalUnit,
+        plain_relative_to: Option<&Date<C>>,
+        context: &mut C::Context,
+    ) -> TemporalResult<Self> {
+        // 1. Assert: If plainRelativeTo is not undefined, calendarRec is not undefined.
+        let plain_relative = plain_relative_to.expect("plainRelativeTo must not be undefined.");
+
+        // 2. Let defaultLargestUnit be DefaultTemporalLargestUnit(years, months, weeks, days, 0, 0, 0, 0, 0).
+        let default_largest = self
+            .iter()
+            .enumerate()
+            .find(|x| x.1 != 0.0)
+            .map(|x| TemporalUnit::from(10 - x.0))
+            .unwrap_or(TemporalUnit::Nanosecond);
```

Comment:
Seems weird to have `Duration` implement `IntoIterator` only for this. Maybe we can just do a manual check of all its components instead?

Reply:
The `IntoIterator` implementation is also used to collect the fields and check the duration's sign and it's validity. This could be refactored to manually check the fields instead of the `IntoIterator` implementation, but I figured using an iterator would be a bit more ergonomic.

That being said, if we were to keep the `IntoIterator` impl, I'm currently wondering whether it should be public.

Reply:
Yeah, the problem is more that we're exposing implementations that are only used internally; it doesn't make much sense to implement IntoIterator for a Duration type. You could also extract that into a function, but I figured it would be easier to just have a `largest_unit` method instead.

Reply:
Removed the `IntoIterator` implementations in favor of an internal `fields` method to return the fields in a `Vec` and did a little clean up to the API along with it to pull back a bit on what we'd be exposing.